keyboards/dmqdesign/spin/keymaps/encoderLayers/keymap.c

```diff
@@ -0,0 +1,148 @@
+/* Copyright 2019-2020 DMQ Design
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "midi.h"
+#include "qmk_midi.h"
+
+#define _BL 0
+#define _FL 1
+#define _TL 15
+
+uint8_t currentLayer;
+
+//The below layers are intentionally empty in order to give a good starting point for how to configure multiple layers.
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_BL] = LAYOUT(/* Base */
+                KC_KP_7, KC_KP_8, KC_KP_9, TO(_BL), \
+                KC_KP_4, KC_KP_5, KC_KP_6, TO(_FL), \
+                KC_KP_1, KC_KP_2, KC_KP_3, TO(_TL), \
+                KC_KP_0, RGB_TOG, RGB_MOD   \
+                ),
+
+    [_FL] = LAYOUT(/* Base */
+                KC_NO, KC_NO, KC_NO, KC_TRNS,\
+                KC_NO, KC_NO, KC_NO, KC_TRNS,\
+                KC_NO, KC_NO, KC_NO, KC_TRNS, \
+                KC_MS_BTN1, KC_NO, KC_MS_BTN2 \
+                ),
+
+    [_TL] = LAYOUT(/* Base */
+                KC_NO, KC_NO, KC_NO, KC_TRNS,\
+                KC_NO, KC_NO, KC_NO, KC_TRNS,\
+                KC_NO, KC_NO, KC_NO, KC_TRNS,\
+                KC_NO, KC_NO, KC_NO  \
+                ),      
```

Comment:
```suggestion
                KC_KP_7, KC_KP_8, KC_KP_9, TO(_BL),
                KC_KP_4, KC_KP_5, KC_KP_6, TO(_FL),
                KC_KP_1, KC_KP_2, KC_KP_3, TO(_TL),
                KC_KP_0, RGB_TOG, RGB_MOD
                ),

    [_FL] = LAYOUT(/* Base */
                KC_NO, KC_NO, KC_NO, KC_TRNS,
                KC_NO, KC_NO, KC_NO, KC_TRNS,
                KC_NO, KC_NO, KC_NO, KC_TRNS,
                KC_MS_BTN1, KC_NO, KC_MS_BTN2
                ),

    [_TL] = LAYOUT(/* Base */
                KC_NO, KC_NO, KC_NO, KC_TRNS,
                KC_NO, KC_NO, KC_NO, KC_TRNS,
                KC_NO, KC_NO, KC_NO, KC_TRNS,
                KC_NO, KC_NO, KC_NO
                ),      
```
You can remove the trailing backslashes here for a cleaner look, if you'd like.